read-fonts/src/tables/variations.rs

```diff
@@ -449,18 +456,26 @@ impl Iterator for DeltaRunIter<'_> {
         //TODO: create a font with data of this shape and go crash some font parsers
         while self.remaining == 0 {
             let control: u8 = self.cursor.read().ok()?;
-            self.are_zero = (control & DELTAS_ARE_ZERO) != 0;
-            self.two_bytes = (control & DELTAS_ARE_WORDS) != 0;
+            // if the top two bits of the control byte (DELTAS_ARE_ZERO and DELTAS_ARE_WORDS) are both set,
+            // then the following values are 32-bit.
+            // <https://github.com/harfbuzz/boring-expansion-spec/blob/main/VARC.md#tuplevalues>
+            let are_zero = (control & DELTAS_ARE_ZERO) != 0;
+            let are_words = (control & DELTAS_ARE_WORDS) != 0;
+            self.value_type = match (are_zero, are_words) {
+                (false, false) => DeltaRunValueType::I8,
+                (false, true) => DeltaRunValueType::I16,
+                (true, false) => DeltaRunValueType::Zero,
+                (true, true) => DeltaRunValueType::I32,
+            };
```

Comment:
maybe this can be a constructor on `DeltaRunValueType`?

Reply:
That would be much nicer! Will do